plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/version/KubernetesVersionManagerImpl.java

```diff
@@ -160,6 +164,33 @@ private List <KubernetesSupportedVersionVO> filterKubernetesSupportedVersions(Li
         return versions;
     }
 
+    private GetUploadParamsResponse registerKubernetesVersionIsoForUpload(final Long zoneId, final String versionName, final String isoChecksum) {
+        CallContext.register(CallContext.current(), ApiCommandResourceType.Iso);
+        String isoName = String.format("%s-Kubernetes-Binaries-ISO", versionName);
+        GetUploadParamsForIsoCmd uploadIso = new GetUploadParamsForIsoCmd();
+        uploadIso = ComponentContext.inject(uploadIso);
+        uploadIso.setName(isoName);
+        uploadIso.setPublicIso(true);
+        if (zoneId != null) {
+            uploadIso.setZoneId(zoneId);
```

Comment:
zoneId - seems to be a mandatory param for upload iso, so I wonder if this could cause some failure if upload iso is done by a user / dom admin 
considering https://github.com/apache/cloudstack/blob/main/server/src/main/java/com/cloud/template/TemplateAdapterBase.java#L194-L197

Reply:
I also need to check if we allow domain admin or a user to add kubernetes versions.